crates/ra_ide_api/src/display/navigation_target.rs

```diff
@@ -115,35 +134,54 @@ impl NavigationTarget {
         let name = module.name(db).map(|it| it.to_string().into()).unwrap_or_default();
         match source {
             ModuleSource::SourceFile(node) => {
-                NavigationTarget::from_syntax(file_id, name, None, node.syntax())
-            }
-            ModuleSource::Module(node) => {
-                NavigationTarget::from_syntax(file_id, name, None, node.syntax())
+                NavigationTarget::from_syntax(file_id, name, None, node.syntax(), None, None)
             }
+            ModuleSource::Module(node) => NavigationTarget::from_syntax(
+                file_id,
+                name,
+                None,
+                node.syntax(),
+                node.doc_comment_text(),
+                node.description(),
+            ),
         }
     }
 
     pub(crate) fn from_module_to_decl(db: &RootDatabase, module: hir::Module) -> NavigationTarget {
```

Comment:
(not in this PR) we really should add `AsNavigationTarget` trait instead of these factory methods